lib/explorer/backend/lazy_frame.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Explorer.Backend.LazyFrame do
+  @moduledoc """
+  Represents a dataframe in an opaque lazy format.
```

Comment:
You can add more information that this is automatically created by functions like `filter_with`, `mutate_with`, etc and it is used to build dataframe expressions.